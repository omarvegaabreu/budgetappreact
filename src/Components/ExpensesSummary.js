import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses-selectors";
import expensesTotal from "../selectors/expensesTotal-selectors";

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? "expense" : "Expenses";
  const formattedExpenseTotal = numeral(expensesTotal / 100).format("$0,0.00");

  return (
    <div>
      <h1>
        Viewing {expenseCount} {expenseWord} totaling: {formattedExpenseTotal}{" "}
      </h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: expensesTotal(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
