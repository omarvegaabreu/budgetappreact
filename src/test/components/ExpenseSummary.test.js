import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("should render 1 expense summary component", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={195} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render multiple expense summary component ", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={51} expensesTotal={654654654321} />
  );
  expect(wrapper).toMatchSnapshot();
});
