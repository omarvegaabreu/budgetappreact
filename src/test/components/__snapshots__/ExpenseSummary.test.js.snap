// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1 expense summary component 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling: 
    $1.95
     
  </h1>
</div>
`;

exports[`should render multiple expense summary component  1`] = `
<div>
  <h1>
    Viewing 
    51
     
    Expenses
     totaling: 
    $6,546,546,543.21
     
  </h1>
</div>
`;
